### PR TITLE
testing for local dependent packages

### PR DESCRIPTION
**Pull request by ckingdon95**

### CODE REVIEW

File: REQUIRE [hunk 1]
```diff
@@ -6,4 +6,4 @@ Documenter 0.7.1
 NamedArrays 0.5.3
 Plots 0.9.4
 JSON
-
+InfoZIP
```
Comment (davidanthoff) [inline REQUIRE]: I'd prefer if this dependency was not listed here. For normal use of Mimi one doesn't need this package, really just for testing, which is probably a very small subset of people that will install Mimi. The previous version had this ``Pkg.installed("ZipFile")`` code at the top of ``test_dependencies.jl``, could you just add that back in but modify it to install InfoZIP if needed?

File: test/test_dependencies.jl [hunk 2]
```diff
@@ -1,76 +1,64 @@
-if Pkg.installed("ZipFile") == nothing
-    Pkg.add("ZipFile")
-end
```
Comment (davidanthoff) [inline test/test_dependencies.jl]: Reenable that, but for InfoZIP.

File: test/test_dependencies.jl [hunk 3]
```diff
@@ -1,76 +1,64 @@
...
-    run(`julia $process`)
-  catch e
-    append!(errors, [(package_name, e)])
-  end
-  #delete current package before testing next one
-  rm(joinpath(tmp_path, package_name), recursive=true)
-end
+    #loop through each dependent package
+    for d in dependencies
+        if  isa_url(d) #what's the best way to know if it's a url?
```
Comment (davidanthoff) [inline test/test_dependencies.jl]: Can you remove the comment here? That is sorted out now.

File: test/test_dependencies.jl [hunk 4]
```diff
@@ -1,76 +1,64 @@
...
+            #find the name of the unzipped package (this only works if the zip archive only has one directory, the package)
+            package_name = readdir(tmp_path)[1]
+            process = string(tmp_path, package_name, "/test/runtests.jl")
+        else
+            package_name = basename(d)
+            process = string(d, "/test/runtests.jl")
+        end
+        #test the package
+        try
+            run(`julia $process`)
```
Comment (tkelman) [inline test/test_dependencies.jl]: `$JULIA_HOME/julia $process`

Comment (davidanthoff) [inline test/test_dependencies.jl]: Actually surprising that this worked so far...

**Pull request merged**